src/main/java/org/reaktivity/specification/socks/internal/SocksFunctions.java

```diff
@@ -126,6 +227,54 @@ public String getPrefixName()
         }
     }
 
+    private static void fillInBytes(
+        byte[] addressBytes,
+        int index,
+        String ipv6Group)
+    {
+        index = 2 * index + 1;
+        for (int i = ipv6Group.length(); i > 0; i-=2)
```

Comment:
the for loop algorithm, the problem is that, it still check `i - 2 > 0` to ensure the substring has a length of 2.